test/server/listener_manager_impl_test.cc

```diff
@@ -406,6 +406,58 @@ TEST_F(ListenerManagerImplTest, AddListenerAddressNotMatching) {
   EXPECT_CALL(*listener_foo, onDestroy());
 }
 
+// Make sure that a listener is created on IPv4 ony setups.
+TEST_F(ListenerManagerImplTest, AddListenerOnIpv4OnlySetups) {
+  InSequence s;
+
+  NiceMock<Api::MockOsSysCalls> os_sys_calls;
+  TestThreadsafeSingletonInjector<Api::OsSysCallsImpl> os_calls(&os_sys_calls);
+
+  ON_CALL(os_sys_calls, socket(AF_INET6, _, 0)).WillByDefault(Return(-1));
+
+  const std::string listener_foo_json = R"EOF(
+  {
+    "name": "foo",
+    "address": "tcp://127.0.0.1:1234",
+    "filters": [],
+    "drain_type": "default"
+  }
+  )EOF";
+
+  ListenerHandle* listener_foo = expectListenerCreate(false);
+  EXPECT_CALL(listener_factory_, createListenSocket(_, _, true));
+
+  EXPECT_TRUE(manager_->addOrUpdateListener(parseListenerFromJson(listener_foo_json), "", true));
+  checkStats(1, 0, 0, 0, 1, 0);
+  EXPECT_CALL(*listener_foo, onDestroy());
+}
+
+// Make sure that a listener is created on IPv6 ony setups.
+TEST_F(ListenerManagerImplTest, AddListenerOnIpv6OnlySetups) {
+  InSequence s;
+
+  NiceMock<Api::MockOsSysCalls> os_sys_calls;
+  TestThreadsafeSingletonInjector<Api::OsSysCallsImpl> os_calls(&os_sys_calls);
+
+  ON_CALL(os_sys_calls, socket(AF_INET, _, 0)).WillByDefault(Return(-1));
+
+  const std::string listener_foo_json = R"EOF(
+  {
+    "name": "foo",
+    "address": "tcp://127.0.0.1:1234",
+    "filters": [],
+    "drain_type": "default"
+  }
+  )EOF";
+
+  ListenerHandle* listener_foo = expectListenerCreate(false);
+  EXPECT_CALL(listener_factory_, createListenSocket(_, _, true));
+
+  EXPECT_TRUE(manager_->addOrUpdateListener(parseListenerFromJson(listener_foo_json), "", true));
```

Comment:
Shouldn't this fail? The address is ipv4, and this test doesn't support ipv4.

Reply:
I should have clarified, I am not testing the "actual" listener creation functionality here, it may fail in the actual setup. But my focus was to test convertDestinationIPsMapToTrie function does not throw exceptions and thus cause listener creation failure of there are no filterMatches defined. I will add this commentary on the test.

Reply:
Ok, I understand better now.

I still think this test case is odd, because this configuration is not supposed to work. I understand that the test still succeeds because it doesn't complete the creation of the listener, but I think it would be better for this one if you change the listener address in the config snippet to a v6 address.

Reply:
Ok. Changed, Please check